### PR TITLE
fix(httpClient): Make . optional in host regex

### LINKED ISSUE #56: Custom hosts are matched against a very strict regex
In Contentful.js there is an option to specify a proxy `host` as described here: https://contentful.github.io/contentful.js/contentful/5.1.3/contentful.html#.createClient

However there is a test against this provided host in this repo here: https://github.com/contentful/contentful-sdk-core/blob/master/lib/create-http-client.js#L9

This regex does not match things like the following:

- localhost
- localhost:4444

This makes it very difficult to do testing against a locally running version of a proxy. Can this be changed to be a bit more lenient?

### PR DESCRIPTION
**Pull request by phoebeschmidt**
Fixes #56

### CODE REVIEW

File: lib/create-http-client.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { isNode, getNodeVersion } from './utils'
 
 // Matches 'sub.host:port' and extracts hostname and port
 // Also enforces toplevel domain specified, no spaces and no protocol
-const HOST_REGEX = /^(?!\w+:\/\/)([^\s:]+\.[^\s:]+)(?::(\d+))?(?!:)$/
+const HOST_REGEX = /^(?!\w+:\/\/)([^\s:]+\.?[^\s:]+)(?::(\d+))?(?!:)$/
```
Comment (makinwab) [inline lib/create-http-client.js:9]: What does this match optionally? My regex not good.

Comment (makinwab) [reply lib/create-http-client.js:9]: Does the comment above still apply based the new regex?

Comment (phoebeschmidt) [reply lib/create-http-client.js:9]: The period should be optional (when it's mandatory, you can't set your host = `localhost`, for example). I tested it with an online regex checker, but we could also surround the period with [] if that's clearer?

Comment (phoebeschmidt) [reply lib/create-http-client.js:9]: my regex are also not 💯

Comment (phoebeschmidt) [reply lib/create-http-client.js:9]: also good point-- I'll modify the comment.

**Pull request merged**